Assets/Scripts/Environmental/Room/RoomScript.cs

```diff
@@ -115,50 +114,83 @@ void LockAllDoors()
     }
 
 
-    /* Loops through all live enemies in the room and dispatches them to go follow
-     * the player.
-     */
-    void DispatchEnemiesToFollowPlayer()
+    /* Plays the associated audio theme attached to the room object.
+     */  
+    private void PlayThemeMusic()
     {
-        foreach (GameObject enemyObject in spawnedEnemies)
+        if (audioTheme != null && audioTheme.clip != null && !audioTheme.isPlaying)
         {
-            Enemy enemy = enemyObject.gameObject.GetComponent<Enemy>();
+            audioTheme.Play();
+        }
+    }
 
-            if (enemy != null)
-            {
-                // The enemy's update loop condition will then be true
-                // enemy.SetMovement(true); // TODO re-enable once implemented
-            }
+
+    /* Stops the associated audio theme attached to the room object.
+     */ 
+    private void StopThemeMusic()
+    {
+        // Stop the theme music for the current room, if there is one
+        if (audioTheme != null && audioTheme.clip != null)
+        {
+            audioTheme.Stop();
         }
     }
 
 
     /* Called when any other collision object enters this Room. Used to detect when the player
-     * enters the room. If there are currently enemies, it will lock all doors.
+     * enters the room. If the player is entering the room for the first time and there are
+     * enemies to spawn, then the room spawns them. Also regulates door locking/unlocking.
      */
-    private void OnTriggerEnter(Collider c)
+    private void OnTriggerEnter(Collider other)
     {
-        // When the player enters this room
-        if (c.gameObject.tag == "Player")
+        // We only care about the player entering the room
+        if (other.gameObject.tag == "Player")
```

Comment:
Just a note: To avoid this method from increasing in width due to nested if statements,  you could employ the [Fail-Fast](https://en.wikipedia.org/wiki/Fail-fast) principle. Specifically, for this player tag conditional, you could just fail fast since no other logic occurs if the Collider is not a player.

Reply:
Gotcha, so return if tag is not Player. Note to self, I'll do that in a cleanup PR.